File: api/Enkl.Api/Controllers/MembersController.cs
using Enkl.Api.Dtos;
using Enkl.Api.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace Enkl.Api.Controllers;

// Every action here is a mutation (members are read via GET /api/projects/{id}'s own project-detail
// graph, not through this controller), so the whole class is Project-Admin-gated — "manage team
// members" is one of the four Project Administrator capabilities.
[ApiController]
[Authorize(Policy = "ProjectAdmin")]
[Route("api/projects/{projectId:guid}/members")]
public class MembersController : ControllerBase
{
    private readonly MemberService _members;

    public MembersController(MemberService members)
    {
        _members = members;
    }

    [HttpPost]
    public async Task<IActionResult> Create(Guid projectId, CreateMemberRequest request)
    {
        var result = await _members.CreateAsync(projectId, request);
        return result is null ? NotFound() : Ok(result);
    }

    [HttpPut("{memberId:guid}")]
    public async Task<IActionResult> Update(Guid projectId, Guid memberId, UpdateMemberRequest request)
    {
        var result = await _members.UpdateAsync(projectId, memberId, request);
        return result is null ? NotFound() : Ok(result);
    }

    [HttpDelete("{memberId:guid}")]
    public async Task<IActionResult> Delete(Guid projectId, Guid memberId)
    {
        return await _members.DeleteAsync(projectId, memberId) ? NoContent() : NotFound();
    }

    // "The project admin role can be assigned to users via the Team management tool" — Project-Admin
    // gated same as every other action here, so only an existing admin can promote/demote another
    // member (ApiValidationException from MemberService.SetProjectAdminAsync's last-admin guard maps
    // to 400 via Program.cs's global exception handler, same as every other manual validation check).
    [HttpPut("{memberId:guid}/admin")]
    public async Task<IActionResult> SetProjectAdmin(Guid projectId, Guid memberId, SetProjectAdminRequest request)
    {
        var result = await _members.SetProjectAdminAsync(projectId, memberId, request.IsProjectAdmin);
        return result is null ? NotFound() : Ok(result);
    }
}

File: php-api/src/Controllers/MembersController.php
<?php

declare(strict_types=1);

namespace Enkl\Api\Controllers;

use Enkl\Api\Db\Database;
use Enkl\Api\Services\MemberService;
use Psr\Http\Message\ResponseInterface as Response;
use Psr\Http\Message\ServerRequestInterface as Request;

/** Ported from Controllers/MembersController.cs. */
final class MembersController extends BaseController
{
    private function service(): MemberService
    {
        return new MemberService(Database::connection());
    }

    public function create(Request $request, Response $response, array $args): Response
    {
        $result = $this->service()->create($args['projectId'], $this->body($request));
        return $result === null ? $this->notFound($response) : $this->json($response, $result);
    }

    public function update(Request $request, Response $response, array $args): Response
    {
        $result = $this->service()->update($args['projectId'], $args['memberId'], $this->body($request));
        return $result === null ? $this->notFound($response) : $this->json($response, $result);
    }

    public function delete(Request $request, Response $response, array $args): Response
    {
        return $this->service()->delete($args['projectId'], $args['memberId']) ? $this->noContent($response) : $this->notFound($response);
    }

    // "The project admin role can be assigned to users via the Team management tool" — Project-Admin
    // gated same as every other action here (routes.php), so only an existing admin can promote/
    // demote another member. MemberService::setProjectAdmin's last-admin guard throws
    // ApiValidationException, mapped to 400 by bootstrap.php's global error handler like every other
    // manual validation check in this tier.
    public function setProjectAdmin(Request $request, Response $response, array $args): Response
    {
        $body = $this->body($request);
        $result = $this->service()->setProjectAdmin($args['projectId'], $args['memberId'], (bool) ($body['isProjectAdmin'] ?? false));
        return $result === null ? $this->notFound($response) : $this->json($response, $result);
    }
}
